### PR TITLE
Guest Reviews

### PR DESCRIPTION
**Pull request by salvoscala**
We need to add guest reviews to RfA. The goal is to let the guests add reviews of the property after a stay and show a list of reviews on the property page.

Add a new Content Type “Review” with:

Title;
Body (as plain text);
Star Rating (use the Fivestar module https://www.drupal.org/project/fivestar);
Reference to Property.

Create a Rule that sends an email to the guest one week after the Checkout. This email (use PET) should contain a message and a link to add a review of the Property (fill the property reference and hide the field).
This rule should be triggered if:

A user books from the site;
A manual booking has been created with a real email (not a dummy email);

Property Page:

In the property page, we should add a list of reviews pulling data from the guest profile (if exists).

We should show a list of reviews (guest name, guest image, title, submission date (with the “time ago” format), trimmed comment and rating (stars)) and the average rating of the property.
We should also add a button with the “Add Review” link if the logged in user is a Guest.

A property owner should be able to add/remove the reviews list from the property page. Let’s add a field (Single On/Off checkbox, enabled by default) to the Roomify Property entities that let the owner either enable or disable reviews for that property page.
He should also be notified (with an email), when a review has been submitted and be able to approve the review. (There's a list of reviews in the dashboard).

### CODE REVIEW

**Comment by salvoscala** [comment]
@acrollet this is ready for CR, when you have a chance. Thanks!

File: modules/roomify/roomify_accommodation_booking/roomify_accommodation_booking.module [hunk 1]
```diff
@@ -3245,6 +3245,24 @@ function roomify_accommodation_booking_notifications_form($form, &$form_state) {
...
+  if (module_exists('roomify_review')) {
+    $form['review'] = array(
+      '#type' => 'fieldset',
+      '#title' => t('Reviews Notification'),
+    );
+
+    $form['review']['roomify_review_notification'] = array(
+      '#type' => 'select',
+      '#title' => t('Review notification'),
+      '#description' => t('How many days after the checkout the request of review should be sent.'),
```
Comment (acrollet) [inline modules/roomify/roomify_accommodation_booking/roomify_accommodation_booking.module]: How about "A request to review the property will be sent this many days after the guest checks out." - unless @cecrs can think of a better way to phrase it?

File: modules/roomify/roomify_review/roomify_review.info [hunk 2]
```diff
@@ -0,0 +1,60 @@
+name = Roomify Review
+description = Reviews feature for RfA
```
Comment (acrollet) [inline modules/roomify/roomify_review/roomify_review.info]: Let's change the description to "Guest reviews for RfA"

File: modules/roomify/roomify_review/roomify_review.module [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+      $image = array(
+        'style_name' => 'square',
+        'path' => $featured_image[0]['uri'],
+      );
+    }
+    $form['property_reviewed']['#weight'] = -10;
+    $form['property_reviewed']['#markup'] = '<div class="roomify-property-reviewed"><div class="featured-image">' . theme('image_style', $image) . '</div><h1>' . t('Review for @name', array('@name' => $property->name)) . '</h1></div>';
+  }
+
+  if (($user->uid != 1) && (!in_array('roomify manager', $user->roles) || !in_array('administrator', $user->roles))) {
```
Comment (acrollet) [inline modules/roomify/roomify_review/roomify_review.module]: we should use "$user->uid !== 1" here

File: modules/roomify/roomify_review/roomify_review.module [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+
+/**
+ * Custom submit function for the review_node_form form.
+ */
+function roomify_review_form_review_node_form_submit($form, &$form_state) {
+  $node = $form_state['node'];
+
+  // Set a message if the review is unpublished.
+  if ($node->status == 0) {
+    drupal_set_message(t('Thanks! Your comment is under review and will be published after a Manager approval.'), 'status');
```
Comment (acrollet) [inline modules/roomify/roomify_review/roomify_review.module]: let's change it to "Thanks! Your review is under moderation and will be published after we've had a chance to approve it."

**Review by acrollet — APPROVED** [review]
looks good! just a few minor changes, approved with these :)

**Pull request merged**